testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/miniapplist/MiniAppListFragment.kt

```diff
@@ -71,26 +80,86 @@ class MiniAppListFragment : BaseFragment(), MiniAppList {
                     Toast.makeText(context, it, Toast.LENGTH_LONG).show()
                 else
                     displayMiniAppList(list)
-                swipeRefreshLayout.isRefreshing = false
+                    downloadedList = list
+                    swipeRefreshLayout.isRefreshing = false
             })
         }
 
-        swipeRefreshLayout.setOnRefreshListener { executeLoadingList() }
+        swipeRefreshLayout.setOnRefreshListener {
+            executeLoadingList()
+            resetSearchBox()
+        }
     }
 
     private fun displayMiniAppList(list: List<MiniAppInfo>) {
         miniAppListAdapter.addListWithSection(list)
+        updateEmptyView(list)
     }
 
     private fun executeLoadingList() {
         viewModel.getMiniAppList()
     }
 
     override fun onMiniAppItemClick(miniAppInfo: MiniAppInfo) {
-        raceExecutor.run { MiniAppDisplayActivity.start(context!!, miniAppInfo) }
+        resetSearchBox()
+        raceExecutor.run {
+            context?.let { MiniAppDisplayActivity.start(it, miniAppInfo) }
+        }
     }
 
     fun switchToInput() {
         raceExecutor.run { activity?.launchActivity<MiniAppInputActivity>() }
     }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        menu.clear()
```

Comment:
What is the use case for menu.clear() in this callback?

Reply:
By default, the search menu is hidden. In this screen, I used `menu.clear()` to re-inflate the whole menu items with adding the search menu as well.

But, actually, I could just inflate the search menu in `onPrepareOptionsMenu` without calling `menu.clear()`. I have updated this part. Thank you for noticing it.